tests/utils/image_builder.py

```diff
@@ -59,13 +60,40 @@ class AgentImageBuilder(object):
 
     IMAGE_TAG = None  # type: six.text_type
     DOCKERFILE = None  # type: Path
+    REQUIRED_IMAGES = []  # type: List[ClassType[AgentImageBuilder]]
 
     # add agent source code to the build context of the image
     COPY_AGENT_SOURCE = False  # type: bool
 
+    # Do not use image cache for this image builder even if 'build' method was called with 'image_cache_path' parameter.
+    # Note. This flag does not affect images in requirements.
+    IGNORE_CACHING = False
+
     def __init__(self):
         self._docker = None  # type: Optional
 
+        # dict with files which need to be copied to build_context.
+        # New paths can be added by using 'add_to_build_context' method.
+        self._things_copy_to_build_context = dict()  # type: Dict[Path, Dict]
+
+        # copy agent course code if needed.
+        if type(self).COPY_AGENT_SOURCE:
+            root_path = Path(get_package_root()).parent
+            self.add_to_build_context(
+                root_path,
+                "agent_source",
+                custom_copy_function=_copy_agent_source
+            )
+
+        # this part iterate through all attributes of the current class and get alla that starts with 'INCLUDE_PATH_'.
+        # the value of this attribute is the path to the file to be copied to the image build context.
+        for name, value in type(self).__dict__.items():
+            if name.startswith("INCLUDE_PATH_"):
```

Comment:
I think it would probably be better to utilize a single class variable to make it a bit less magical and more explicit.


Perhaps ``INCLUDE_PATHS = []``. And if we need access to the name as well, we can either make item a dictionary or tuple.